src/client.rs

```diff
@@ -143,11 +141,12 @@ impl<WorldType: World> Client<WorldType> {
     ///     # break;
     /// }
     /// ```
-    pub fn update<NetworkResourceType: NetworkResource>(
+    pub fn update(
         &mut self,
         delta_seconds: f64,
         seconds_since_startup: f64,
-        net: &mut NetworkResourceType,
+        net: &mut <WorldType::ConfigType as Config>::NetworkResourceType,
+        client_id: <<WorldType::ConfigType as Config>::NetworkResourceType as NetworkResource>::ConnectionHandleType,
```

Comment:
Are we intentionally shifting the responsibility to users of this library to figure out what their `client_id` is?

Previously, the `client_id` was automatically detected by the `ClockSyncer` so that the users of this library won't need to. I suppose, if you're using `SocketAddr`, the clients can figure out their `cliend_id` themselves, but this might not be the case for other `ConnectionHandleType`s. Would it be compatible with your use case if we tried keeping the original auto-detection-of-client-id feature?

Reply:
In the first version of PR this was needed as it could not be determined without external information.
I do see the value of passing it explicitly internally, but I agree it could be removed from external API surface.

Reply:
Doh.
I tried changing that and I need to add `client_id` back to `ClockSyncMessage` - which makes it parametrized type, which ripples through the code even more.